File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.py
class Solution(object):
    def findDuplicate(self, nums):
        tort=nums[0]
        hare=nums[0]
        while True:
            tort=nums[tort]
            hare=nums[nums[hare]]
            if tort==hare:
                break
        tort=nums[0]
        while tort!=hare:
            tort=nums[tort]
            hare=nums[hare]
        return tort

        

File: 0287-find-the-duplicate-number/0287-find-the-duplicate-number.c
int findDuplicate(int* nums, int numsSize) {
    for(int i=0;i<numsSize;i++){
        int pos = abs(nums[i])-1;
        if(nums[pos] < 0) return abs(nums[i]);
        nums[pos] = -nums[pos];
    }
    return 1;
}